### PR TITLE
Enhance file cache to encrypt data written to the cache file

### PR DESCRIPTION
**Pull request by jiangliu**
Enhance the FileCache subsystem to optionally encrypt data written to the local cache file.

### CODE REVIEW

**Comment by jiangliu** [comment]
A sample `nydusd.conf` to test the encryption. You may tweek with ` enable_convergent_encryption` and `enable_encryption`.
```
version = 2
id = "id1"
backend.type = "localfs"
backend.localfs.dir = "/root/ws/image-service.git/images"
cache.type = "filecache"
cache.compressed = false
cache.validate = true
cache.filecache.work_dir = "/root/ws/image-service.git/cache"
cache.filecache.enable_encryption = true
cache.filecache.enable_convergent_encryption = false
cache.filecache.encryption_key = "fc4a7db5614afc2f400e9478bebed1aefdbc9d7cd03210b84f144683a7a6fd1a"
rafs.mode = "direct"
```

**Comment by ccx1024cc** [comment]
> A sample `nydusd.conf` to test the encryption. You may tweek with ` enable_convergent_encryption` and `enable_encryption`.
> 
> ```
> version = 2
> id = "id1"
> backend.type = "localfs"
> backend.localfs.dir = "/root/ws/image-service.git/images"
> cache.type = "filecache"
> cache.compressed = false
> cache.validate = true
> cache.filecache.work_dir = "/root/ws/image-service.git/cache"
> cache.filecache.enable_encryption = true
> cache.filecache.enable_convergent_encryption = false
> cache.filecache.encryption_key = "fc4a7db5614afc2f400e9478bebed1aefdbc9d7cd03210b84f144683a7a6fd1a"
> rafs.mode = "direct"
> ```

Please add a e2e test, instead of manual way~~~~

**Comment by jiangliu** [comment]
> > A sample `nydusd.conf` to test the encryption. You may tweek with ` enable_convergent_encryption` and `enable_encryption`.
> > ```
> > version = 2
> > id = "id1"
> > backend.type = "localfs"
> > backend.localfs.dir = "/root/ws/image-service.git/images"
> > cache.type = "filecache"
> > cache.compressed = false
> > cache.validate = true
> > cache.filecache.work_dir = "/root/ws/image-service.git/cache"
> > cache.filecache.enable_encryption = true
> > cache.filecache.enable_convergent_encryption = false
> > cache.filecache.encryption_key = "fc4a7db5614afc2f400e9478bebed1aefdbc9d7cd03210b84f144683a7a6fd1a"
> > rafs.mode = "direct"
> > ```
> 
> Please add e2e tests, instead of in manual way~~~~

It has long dependency chain to enable unit test for this feature. We need to enhance nydus/snapsotter/converter, nydusify and smoke:) 
So it would be great to merge this first so we can do it step by step.

File: api/src/config.rs [hunk 1]
```diff
@@ -672,6 +672,15 @@ pub struct FileCacheConfig {
...
     pub disable_indexed_map: bool,
+    /// Enable encryption data written to the cache file.
+    #[serde(default)]
+    pub enable_encryption: bool,
+    /// Enable convergent encryption for chunk deduplication.
+    #[serde(default)]
+    pub enable_convergent_encryption: bool,
+    /// Key for data encryption, a heximal representation of [u8; 32].
+    #[serde(default)]
+    pub encryption_key: String,
```
Comment (imeoer) [inline api/src/config.rs:683]: Does this need to be persisted in local disk?

Comment (imeoer) [reply api/src/config.rs:683]: Hope we can pass it by API next.

**Review by imeoer — APPROVED** [review]

**Pull request merged**